cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AbstractMonitorHandler.java

```diff
@@ -226,7 +226,13 @@ private void restartInstances(HttpResponder responder, String serviceName, int i
 
     try {
       if (!masterServiceManager.isServiceEnabled()) {
-        throw new IllegalStateException();
+        LOG.debug(String.format("Fail restart instances because service %s is not enabled.", serviceName));
+
+        isSuccess = false;
+        responder.sendString(HttpResponseStatus.FORBIDDEN,
+                             String.format("Fail to restart instance for % because the service is not enabled.",
+                                           serviceName));
```

Comment:
Why are these messages not the same? I'd go with this instead for each:
```
"Failed to restart instance for % because the service is not enabled.", serviceName
```